account_invoice_triple_discount/models/account_invoice.py

```diff
@@ -32,8 +32,15 @@ class AccountInvoiceLine(models.Model):
         default=0.0,
     )
 
-    @api.multi
-    @api.depends('discount2', 'discount3')
+    # Each iteration of api.multi is done into a different cache!
+    # As we're using cache to define a value into discount field, we have
+    # to stay into the same cache. So we can not use api.multi in this case.
+    @api.one
```

Comment:
Deprecated, see https://github.com/odoo/odoo/blob/feef39aee083294f8c8d19aecef0434af04c48d0/odoo/api.py#L359

Reply:
@SimoRubi Thank you for your comment but I explain why I was forced to use `@api.one` (instead of `multi`) on the line just before.